### PR TITLE
Add initial StripeAttest code

### PR DESCRIPTION
**Pull request by davidme-stripe**
## Summary
Adds StripeAttest, a class for implementing basic App Attestation against the Stripe API.

## Motivation
Enabling basic App Attestation support in the SDK.

## Testing
Added unit tests and additional tests for a few edge cases.

## Changelog
Not user-visible.

### CODE REVIEW

File: StripeCore/StripeCore/Source/Attestation/StripeAttest.swift [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+    }
+
+    private static var keyID: String? {
+        get {
+            UserDefaults.standard.string(forKey: DefaultsKeys.keyID.rawValue)
+        }
+        set {
+            UserDefaults.standard.set(newValue, forKey: DefaultsKeys.keyID.rawValue)
+        }
+    }
```
Comment (mats-stripe) [inline StripeCore/StripeCore/Source/Attestation/StripeAttest.swift:89]: Out of scope, but makes me wonder if we should pull out our `@UserDefaults` [property wrapper](https://github.com/stripe/stripe-ios/blob/master/Example/FinancialConnections%20Example/FinancialConnections%20Example/Playground/helpers/UserDefaults.swift) to StripeCore 👀 😅 

```swift
@UserDefault(DefaultsKeys.keyID.rawValue, defaultValue: nil)
static var keyID: String?
```

**Review by mats-stripe — COMMENTED** [review]
Really nice job documenting all this complex code - it made it very easy to follow along from having nearly no context on how attestation works. The API seems like it will be super easy to integrate in FC! 👏 Happy to ✅ this once it's ready to go!

File: StripeCore/StripeCore/Source/Attestation/StripeAttest.swift [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+            // If error is DCErrorInvalidKey (3) and the domain is DCErrorDomain,
+            // then the key is either unattested or corrupted.
+            let error = error as NSError
+            if error.domain == DCErrorDomain && error.code == DCError.invalidKey.rawValue {
+                if retryIfNeeded {
+                    // We'll try to attest again, maybe our initial attestation was unsuccessful?
+                    // `DCError.invalidKey` could mean a lot of things, unfortunately.
+                    // If this doesn't work, then in `attest()` we'll deem the key to be corrupted
+                    // and throw it out.
+                    try await attest()
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply StripeCore/StripeCore/Source/Attestation/StripeAttest.swift:265]: It isn't a straightforward retry, it's re-attesting the key (in case it wasn't correctly attested) and *then* retrying with the re-attested key or a new key. So a timeout wouldn't change the behavior. I can make that clearer in the comments/naming!

File: StripeCore/StripeCoreTests/Attestation/StripeAttestTest.swift [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        let invalidKeyError = NSError(domain: DCErrorDomain, code: DCError.invalidKey.rawValue, userInfo: nil)
+        // But fail the assertion, causing the key to be reset
+        mockAttestService.shouldFailAssertionWithError = invalidKeyError
+        do {
+            _ = try await stripeAttest.assert()
+            XCTFail("Should not succeed")
+        } catch {
+            // Should get a rate limiting error when we try to generate the second key:
+            XCTAssertEqual(error as! StripeAttest.AttestationError, StripeAttest.AttestationError.keygenRateLimitExceeded)
+        }
```
[the remark this answers is not in the document]
Comment (davidme-stripe) [reply StripeCore/StripeCoreTests/Attestation/StripeAttestTest.swift]: I tried that at first, but it unfortunately doesn't appear to support async functions!

File: StripeCore/StripeCore/Source/Attestation/StripeAttest.swift (hunk 1 above)
Comment (davidme-stripe) [reply StripeCore/StripeCore/Source/Attestation/StripeAttest.swift:89]: Oooh, neat! We should!

File: StripeCore/StripeCoreTests/Attestation/StripeAttestTest.swift (hunk 3 above)
[the remark this answers is not in the document]
Comment (mats-stripe) [reply StripeCore/StripeCoreTests/Attestation/StripeAttestTest.swift]: Ahhh darn!

File: StripeCore/StripeCore/Source/Attestation/StripeAttest.swift (hunk 2 above)
[the remark this answers is not in the document]
Comment (mats-stripe) [reply StripeCore/StripeCore/Source/Attestation/StripeAttest.swift:265]: Ahh gotcha, that makes sense!

File: modules.yaml [hunk 4]
```diff
@@ -155,8 +155,8 @@ modules:
     output: docs/stripeapplepay
     readme: StripeApplePay/README.md
   size_report:
-    max_compressed_size: 500
-    max_uncompressed_size: 1200
+    max_compressed_size: 700
+    max_uncompressed_size: 1700
```
Comment (mats-stripe) [inline modules.yaml:159]: Does this PR increase the `StripeApplePay` pod size by ~40%? 😮

**Review by mats-stripe — APPROVED** [review]

**Comment by davidme-stripe** [comment]
Unfortunately I had to bump up the size of StripeApplePay and StripeFinancialConnections due to the inclusion of `async` support, but I don't think it will cause problems for most users: If the minimum iOS version of their app clip is high enough, Xcode won't bundle  libswift_Concurrency.dylib.

File: modules.yaml (hunk 4 above)
Comment (davidme-stripe) [reply modules.yaml:159]: Yeah, see below comment, this is because of `libswift_Concurrency.dylib` back-deployment. :( Fortunately it should only increase the size for apps targeting older iOS versions, though I'm not sure of the exact versions.

Comment (davidme-stripe) [reply modules.yaml:159]: I also considered putting this somewhere else, but we need to use it in FinancialConnections, Payments, and PaymentSheet, so there isn't a better location for it. We're planning to add more async support in 2025, so I think it's okay to move to it now.
